src/models/dirac_mixture_multi_publisher_model.py

```diff
@@ -0,0 +1,579 @@
+# Copyright 2021 The Private Cardinality Estimation Framework Authors
```

Comment:
Hi Jiayu,

This looks very interesting.  Thank you for writing it.  This is going to be a major effort to review this code, so I would be very grateful if you could break it into two or more PR's.  Also, is there an updated doc that fully describes the current version of the models?  This would be very helpful in reviewing the code.  Thanks!

Matt

Reply:
Leaving a note: following Matt's suggestion, this PR is split into 4 PRs for review, tentatively PR 103, 105, 106, 107.  You may still read this PR for the complete codes, but please go to split PRs for conversations and updates.